src/modules/methods.js

```diff
@@ -0,0 +1,157 @@
+        if (button.id > 0) {
+          id = button.id - 1;
+        } else {
+          id = 0;
+        }
+        this.deleteListData(id);
+        this.showLists();
+      }));
+    };
+
+    // section created dynamiclly
+    static toDoListsHtml = ({ description, index }) => {
+      const ul = document.createElement('ul');
+      ul.className = 'to-do';
+      ul.innerHTML = `
+        <li><input class="checkbox" id="CHECK${index}" type="checkbox"></li> 
+        <li><input id="LIST${index}" type="text" class="text" value="${description}" readonly></li>
+        <li class="edit-remove">
+        <button class="edit_list_btn" id="${index}"><i class="fa fa-ellipsis-v icon"></i></button>
+        <button class="remove_btn" id="${index}"><i class="fa fa-trash-can icon"></i></button>
+        </li>
+      `;
+      return ul;
+    }
+
+    static showLists = () => {
+      const toDoLists = this.getToDoListFromLocalStorage();
+      document.querySelector('.toDoListContainer').innerHTML = '';
+      toDoLists.forEach((item) => {
+        document.querySelector('.toDoListContainer').appendChild(this.toDoListsHtml(item));
+      });
+
+      this.removeToDoListBtn();
+      this.editListBtnEvent();
+      this.updateListBtnEvent();
+    };
+
+    // add a task to a list
+    static addLists = (description) => {
+      const toDoLists = this.getToDoListFromLocalStorage();
+      const index = toDoLists.length + 1;
+      const newtask = new DataList(description, false, index);
+
+      toDoLists.push(newtask);
+      this.addListToStorage(toDoLists);
+      this.showLists();
+    }
+
+    // update to do list
+    static updateListBtnEvent = () => {
+      document.querySelectorAll('.text').forEach((input) => input.addEventListener('keypress', (event) => {
+        if (event.key === 'Enter') {
+          event.preventDefault();
+          const inputListId = 'LIST';
+          const ListIdSelected = event.currentTarget.id;
+          let listID;
+
+          if (!ListIdSelected.includes('LIST')) {
+            listID = inputListId.concat(ListIdSelected);
+          } else {
+            listID = ListIdSelected;
+          }
+
+          document.getElementById(listID).setAttribute('readonly', 'readonly');
+          this.ListInputUpdate(document.getElementById(listID).value, (Number(listID.replace('LIST', '')) - 1));
+        }
+      }));
+    }
+
+    // edit list
+    static editListBtnEvent = () => {
+      document.querySelectorAll('.edit_list_btn').forEach((button) => button.addEventListener('click', (event) => {
+        event.preventDefault();
+        const inputListId = 'LIST';
+        const ListIdSelected = event.currentTarget.id;
+        let listID;
+
+        if (!ListIdSelected.includes('LIST')) {
+          listID = inputListId.concat(ListIdSelected);
+        } else {
+          listID = ListIdSelected;
+        }
+
+        const listItem = event.target.closest('li');
+        const ulItem = event.target.closest('ul');
+
```

Comment:
- [ ] You did great work on this project but there is something you should work on for your project to fit all requirements since we are not able to edit a task on our to-do list and as required for this milestone of your project A function for editing task descriptions should be implemented. when I'm trying to edit a task and refresh the page by editing disappeared :wink:  :upside_down_face:  kindly fix this issue

Reply:
Good day dear code review team, i've recently reached out to @Alfredbis29 regarding the required changes noted above, during our zoom call I had the chance to share my screen and show him the required changes were already implemented and functional as close as possible for this projects milestone requirements, we assumed it might have been a browser error or the webpackage was not loaded fully when he had the chance to test the project, and so I thought taking screen shots of the requirements being present and working in my project already would not do it justice so I took the liberty of creating a [demonstration video linked here](https://drive.google.com/file/d/1tDGbtNbsjyLqoyPulFd6FFlGfnTGuyPr/view?usp=sharing) which shows me explaining that the index order works on both google chrome and Microsoft edge

### issue 1
- the to do list didn't seem to be indexing the listed tasks in the correct order when one of the listed tasks was deleted or when the page was reloaded

### video explanation for issue 1
( **in the video I show how i listed items in the to-do list, and how they are re-ordered  with the index when deleted and remained in that order when the page was refreshed**)

### issue 2
- it seemed as if there was no functionality to edit the already listed tasks, and they would disappear when the page was reloaded

### video explanation for issue 2
(**in the video I add a list of numbers { 1,2,3,4 } to the todo list, refresh the page a few times, and then i spelled out the numbers in words while refreshing the page after i add a new word number**)

Reply:
Thank you for the code review @Alfredbis29 I really do appreciate you taking the time out to answer my zoom call earlier in order for me to show you these issues